src/anet.c

```diff
@@ -251,10 +255,22 @@ int anetResolve(char *err, char *host, char *ipbuf, size_t ipbuf_len,
     hints.ai_family = AF_UNSPEC;
     hints.ai_socktype = SOCK_STREAM;  /* specify socktype to avoid dups */
 
+    if (flags & ANET_PREFER_IPV4) {
+        hints.ai_family = AF_INET;
+        if ((rv = getaddrinfo(host, NULL, &hints, &info)) == 0) goto found;
+        hints.ai_family = AF_INET6;
+    } else if (flags & ANET_PREFER_IPV6) {
+        hints.ai_family = AF_INET6;
+        if ((rv = getaddrinfo(host, NULL, &hints, &info)) == 0) goto found;
+        hints.ai_family = AF_INET;
+    }
+
     if ((rv = getaddrinfo(host, NULL, &hints, &info)) != 0) {
         anetSetError(err, "%s", gai_strerror(rv));
         return ANET_ERR;
     }
+
+found:
```

Comment:
Just and idea to avoid goto. I don't know if you like it.

```suggestion
    if (flags & ANET_PREFER_IPV4) {
        hints.ai_family = AF_INET;
    } else if (flags & ANET_PREFER_IPV6) {
        hints.ai_family = AF_INET6;
    }

    rv = getaddrinfo(host, NULL, &hints, &info);
    if (rv != 0 && hints.ai_family != AF_UNSPEC) {
        /* Try the other IP version. */
        hints.ai_family = (hints.ai_family == AF_INET) ? AF_INET6 : AF_INET;
        rv = getaddrinfo(host, NULL, &hints, &info);
    }
    if (rv != 0) {
        anetSetError(err, "%s", gai_strerror(rv));
        return ANET_ERR;
    }
```

Reply:
thanks, it looks good, took it